arangod/Agency/Node.cpp

```diff
@@ -1196,22 +1267,22 @@ Slice Node::getArray() const {
     throw StoreException("Not an array type");
   }
   rebuildVecBuf();
-  return Slice(_vecBuf.data());
+  return Slice(_vecBuf->data());
 }
 
 void Node::clear() {
-  _children.clear();
+  _children.reset();
   removeTimeToLive();
-  _value.clear();
-  _vecBuf.clear();
+  _value.reset();
+  _vecBuf.reset();
```

Comment:
Not sure how often this is called, but instead of `reset`ing `_value` and `_children`, we may also call `->clear()` on them. This will leave the objects in place with empty vectors, so the vectors can be recycled later if the objects are refilled.
But I can't tell if that is a frequent use case and worth it.

Reply:
I think it can happen that a particular `Node` object switches between being a JSON object (in which case it allocates `_children` unless it is empty) and something else (JSON array or other value) but I do not expect it to happen very frequently. Therefore I have chosen to always clear up all possible allocations.